adr/ADR-8.md

```diff
@@ -216,7 +216,11 @@ Writing a key to the bucket is a basic JetStream request.
 The KV key `auth.username` in the `CONFIGURATION` bucket is written sent, using a request, to `$KV.CONFIGURATION.auth.username`.
 
 To implement the feature that would accept a write only if the sequence of the current value of a key has a specific sequence
-we use the new `Nats-Expected-Last-Subject-Sequence` header.
+we use the new `Nats-Expected-Last-Subject-Sequence` header. Since a recent server version the special value `0` for this header
+would indicate that the message should only be accepted if it's the first message on a subject. This is purge aware, ie. if 
+a value is in and the subject is purged again a `0` value will be accepted.
+
+This can be implemented as a `PutOption` ie. `Put("x.y", []byte("hello world"), OnlyIfLastValueSequence(10))`.
```

Comment:
I kinda like Create() here tbh.

Reply:
I support the idea of having `Put` as the minimum but `Create` and `Update` can be coded easily enough with Get (Entry) first or put with sequence expectations. 
The problem I do see, is who decides (the user or us) whether Create is valid if the key was deleted as well as has has never existed. Similar with update. So that behavior would have to be defined.
Looking further down in the changes, this could be accomplished by the options versions of the put. The options need to be defined.